### DISCUSSION TITLE
Show days that can be skipped in advance

### DISCUSSION THREAD

**Discussion by jwleon** [iSoron/uhabits#372] [Feature Requests]
Hello!

I have a feature suggestion: For habits that are not all days of a week, it would be nice to see, if the next days are currently backed by the x of y "policy" and can be skipped without breaking the streak. This would save me headache of figuring it out for the next day (which is, in some cases, not too easy).

Thanks in advance for reading and maybe considering it and thanks for this great app! :-)

**Comment by vyu1** [comment]
I second this. I've also personally found this frustrating

**Comment by Anshoe** [comment]
That would be quite useful

**Comment by hiqua** [comment]
Would be nice to have.

Ideally we could scroll sideways to show more days, including days in the future, where the marks would also be filled.

**Reply by hiqua** [reply]
In the meantime I discovered that we can actually scroll in the past by swiping the dates. I think it would be enough to allow users to swipe in the other direction, and ~~the code changes should be minimal~~

**Comment by dkuku** [comment]
also would be nice to have a way to skip days in advance. sometimes you know that you won't be able to do something because of family meeting/travelling or 1000 other things

**Comment by drawtheliner** [comment]
Yes, please add this!

(I like how consistent - and subtle - the comments on this feature are - almost one comment per year since 2018!)

**Comment by TAHayden** [comment]
It does this now, right?

**Reply by drawtheliner** [reply]
Not that I'm aware of.

You are able to scroll left (ie., show days in the future)?

**Reply by TAHayden** [reply]
Ah, gotcha. You're quite right, of course! (I hadn't had enough coffee!)

**Comment by KosmicGR** [comment]
Technically streaks show the beginning and end of their streak, but if the streak is not visible because it's not as big as the others, that information is unavailable.

For example, Let's say you have a habit that needs to be done _every 14 days_ and you've done it on the 30th of September.
It will mention the end of the streak being the 13th of October, meaning you'll have to do it again on the 14th.

I do agree this information should be more reliably visible.

**Comment by JetFlightGit** [comment]
I know it's been ages since anything got posted here but I am really looking for that Feature.

I would like to rather do some Tasks in advance then too late.

Is there still Updates coming? Probably only folrom the community maybe

**Comment by ezio-melotti** [comment]
I would also find this useful for two reasons:
1. For daily habits (e.g. exercises), sometimes I know I won't be able to do them tomorrow or the day after, so I do them in advance and I would like to mark them on the correct (future) day (e.g. if tomorrow I can't do 50 pushups, I do 100 today and I want to mark 50 for today and 50 for tomorrow);
2. For "N times per week" habits, I'd like to see when I'm supposed to do it next so I can plan my week better (what the OP requested);

Implementation wise, I would be happy with just being able to scroll sideways on the dates in the main screen as it was suggested above (i.e. scroll left to see the past, scroll right to see the future).  Personally, being able to see a week in the future would be enough, but other users might need more.

This feature has also been request in these posts:
- https://github.com/iSoron/uhabits/discussions/2042
- https://github.com/iSoron/uhabits/discussions/1160
- https://github.com/iSoron/uhabits/discussions/1998

**Reply by KyleSCraig** [reply]
It seems potentially helpful to mention that what I see as the biggest complication of scrolling to the future is when you use the "hide entered" setting.
